lib/transports.js

```diff
@@ -6,12 +6,16 @@ var util = require('util');
 function Transport() {}
 util.inherits(Transport, events.EventEmitter);
 
+var agentOptions = { keepalive: true, maxSockets: 100 };
```

Comment:
100 seems high to me? Also, should we provide a `maxFreeSockets` option? (https://nodejs.org/api/http.html#http_new_agent_options)

Reply:
Number of `maxFreeSockets` won't go over `maxSockets` number and it defaults to `256` so we should be good here.